src/options-resolver.ts

```diff
@@ -15,33 +15,89 @@
+import {Validator} from './validators/validator'
 
-import * as readlineSync from 'readline-sync'
-
-export const OptionsResolver = (options: any,
-                                key: string,
-                                secondSource: () => string | undefined,
-                                promptText: string,
-                                readlineDependency?: any,
-                                hide?: boolean): any => {
-    const readline = readlineDependency || readlineSync
-    const hideEchoBack = hide ? true : false
-    return options[key] !== undefined ? options[key] : (secondSource() ||
-        readline.question(promptText, {hideEchoBack}))
+export const OptionsChoiceResolver = async (options: any,
+                                            key: string,
+                                            promptText: string,
+                                            choices: Choices[],
+                                            type: SelectOptionType = 'select',
+                                            validation: Validator<any> | undefined) => {
+    if (!['select', 'multiselect'].includes(type)) {
+        console.log(chalk.red('ERR'), 'Invalid options choice resolver type')
+        return process.exit()
+    }
+    if (options[key] !== undefined) {
+        const title = options[key].trim()
+        if (validation !== undefined ) {
+            const test = validation.validate(title)
+            if (typeof test === 'string') {
+                console.log(chalk.red('ERR'), test)
+                return process.exit()
+            }
+        }
+        return choices.find((item) => item.title === title)?.value
+    } else {
+        return (await prompts({
+            type,
+            name: key,
+            message: promptText,
+            choices,
+            validate: validation !== undefined ?
+                (result) => validation.validate(result) : () => true,
+        }))[key]
+    }
 }
 
-export const OptionsChoiceResolver = (options: any,
+export const OptionsResolver = async (options: any,
                                       key: string,
+                                      secondSource: () => string | undefined,
                                       promptText: string,
-                                      choices: string[],
-                                      readlineDependency?: any): any => {
-    const readline = readlineDependency || readlineSync
-    if (options[key] !== undefined) {
-        return options[key]
-    }
-    const choiceIndex = readline.keyInSelect(choices, promptText)
-    if (-1 === choiceIndex) {
+                                      type: InputOptionType = 'text',
+                                      validation: Validator<any> | undefined) => {
+
+    if (!['text', 'password', 'number'].includes(type)) {
+        console.log(chalk.red('ERR'), 'Invalid options resolver type')
         return process.exit()
     }
```

Comment:
Maybe we could remove these codes. Because I defined a customized type `SelectOptionType` to limit this value.
[src/interfaces/options.interface.ts](https://github.com/nemtech/symbol-cli/blob/56be4409bcc67d16278659e811871ef216f27bdb/src/interfaces/options.interface.ts)

Reply:
Solved :+1: